test/test_mysql2.rb

```diff
@@ -0,0 +1,148 @@
+require 'test/unit'
+require 'semian/mysql2'
+require 'toxiproxy'
+require 'timecop'
+
+class TestMysql2 < Test::Unit::TestCase
+  def setup
+    @proxy = Toxiproxy[:semian_test_mysql]
+    Semian.destroy(:mysql_testing)
+  end
+
+  def teardown
+    threads.each { |t| t.kill }
+    @threads = []
+  end
+
+  def test_semian_identifier
+    assert_equal :mysql_foo, FakeMysql.new(semian: {name: 'foo'}).semian_identifier
```

Comment:
Any major reason you didn't want to use the MySQL client? I don't see a reason with connecting N times here.